commands/build_bundles.py

```diff
@@ -0,0 +1,202 @@
+            zip_file.writestr(filename, filecontent)
+    with open(output_path, "wb") as f:
+        f.write(zip_buffer.getvalue())
+    print("Wrote %r" % output_path)
+
+
+def sync_cloud_storage(
+    storage_bucket: str, remote_folder: str, to_upload: list[str], to_delete: list[str]
+):
+    """
+    Upload the specified `to_upload` filenames, and delete the specified `to_delete` filenames
+    from the `remote_folder` of the `storage_bucket`.
+    """
+    # Ensure you have set the GOOGLE_APPLICATION_CREDENTIALS environment variable
+    # to the path of your Google Cloud service account key file before running this script.
+    client = storage.Client()
+    bucket = client.bucket(storage_bucket)
+    for filename in to_upload:
+        remote_file_path = os.path.join(remote_folder, filename)
+        blob = bucket.blob(remote_file_path)
+        blob.upload_from_filename(filename)
+        print(f"Uploaded {filename} to gs://{storage_bucket}/{remote_file_path}")
+
+    to_delete = {os.path.join(remote_folder, f) for f in to_delete}
+    blobs = bucket.list_blobs(prefix=remote_folder)
+    for blob in blobs:
+        if blob.name in to_delete:
+            blob.delete()
+            print(f"Deleted gs://{storage_bucket}/{blob.name}")
+
+
+def build_bundles(event, context):
+    """
+    Main command entry point that:
+    - fetches all collections changesets
+    - builds a `changesets.zip`
+    - fetches attachments of all collections with bundle flag
+    - builds `{bid}--{cid}.zip` for each of them
+    - send the bundles to the Cloud storage bucket
+    """
+    rs_server = event.get("server") or SERVER
+
+    client = KintoClient(server_url=rs_server)
+
+    base_url = client.server_info()["capabilities"]["attachments"]["base_url"]
+
+    existing_bundle_timestamp = get_modified_timestamp(
+        f"{base_url}{DESTINATION_FOLDER}/changesets.zip"
+    )
+    if existing_bundle_timestamp is None:
+        print("No previous bundle found")  # Should only happen once.
+        existing_bundle_timestamp = -1
+
+    all_changesets = fetch_all_changesets(client)
+    highest_timestamp = max(c["timestamp"] for c in all_changesets)
+
+    if existing_bundle_timestamp >= highest_timestamp:
+        print("Existing bundle up-to-date. Nothing to do.")
+        return
+
+    bundles_to_upload = []
+    bundles_to_delete = []
+
+    write_zip(
+        "changesets.zip",
+        [
+            ("{bucket}--{metadata[id]}.json".format(**changeset), json.dumps(changeset))
+            for changeset in all_changesets
+        ],
+    )
+    bundles_to_upload.append("changesets.zip")
+
+    for changeset in all_changesets:
+        bid = changeset["bucket"]
+        cid = changeset["metadata"]["id"]
+        should_bundle = changeset["metadata"].get("attachment", {}).get("bundle", False)
+        attachments_bundle_filename = f"{bid}--{cid}.zip"
+
+        if not should_bundle:
+            bundles_to_delete.append(attachments_bundle_filename)
```

Comment:
Nice, doing all the writes at the end later will prevent some possible contention issues.